### PR TITLE
Updates to Fringe Rate Filtering and Reflection Fitting

### PR DESCRIPTION
**Pull request by nkern**
Updates to Fringe-Rate Filtering object by adding the PAPER technique of applying an FIR filter to the data.

Updates to Reflection Fitter object by adding a `refine_auto_cal` method that sets up a loss function and perturbs the reflection parameters to get a refined fit

Minor updates to cross talk modeling API

Also a small bug fix to `HERAData.select` w/ `inplace` kwarg

### CODE REVIEW

**Comment by nkern** [comment]
@jsdillon not sure what's going on but your new `test_redcal_post_abscal` test is only failing on python3 environment. lmk if you have any insights..

**Comment by jsdillon** [comment]
@nkern That's super wierd, especially since your the travis PR passed and the travis push failed. I'll just restart the job.

**Comment by jsdillon** [comment]
@nkern after two restarts, it passed. Who knows?

File: hera_cal/reflections.py [hunk 1]
```diff
@@ -373,18 +539,12 @@ def pca_decomp(self, dfft, dly_range, flags=None, side='both', keys=None, Nkeep=
...
-            else:
-                Nmodes = Nkeep
+            u, svals, v = np.linalg.svd(dfft[k] * wgts[k], full_matrices=False)
 
             # append to containers: keeping Nmodes b/c otherwise their eventual outerproduct fails
-            self.umodes[k] = u[:, :Nmodes]
-            self.vmodes[k] = v[:Nmodes, :]
-            self.svals[k] = svals
+            self.umodes[k] = u[:, :Nkeep]
+            self.vmodes[k] = v[:Nkeep, :]
```
[the remark this answers is not in the document]
Comment (nkern) [reply hera_cal/reflections.py]: That is handled by the `svd(... full_matrices=False)`, which truncates either `u` or `v` depending on which has more modes to begin with. Therefore, if `Nkeep > Ndelay > Ntimes` then effectively `Nkeep = Ndelay` or if `Nkeep > Ntimes > Ndelay` then effectively `Nkeep = Ntimes`.

**Comment by nkern** [comment]
Okay, this stuff is going in! LMK if anyone has more questions / issues with this code and we can chat later!

**Pull request merged**